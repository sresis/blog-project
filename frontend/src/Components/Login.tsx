import React, {useState, createContext } from 'react';
import axios from 'axios'
import { useHistory } from 'react-router-dom';
import { couldStartTrivia } from 'typescript';
import {Card } from 'react-bootstrap';
const initialInputs = {
    username: '',
    password: ''
  }

function Login () {
    const history = useHistory()
    const [loginInput, setLoginInput] = useState<{username: string; password: string;}>(initialInputs)

    // handle when user types login info
    const handleLoginInput = (event: React.ChangeEvent<HTMLInputElement>) => {
        setLoginInput({
          ...loginInput,
          [event.currentTarget.name]: event.currentTarget.value
        })
      }
    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    axios.post('http://localhost:8080/login', loginInput)
    .then((res: any) => {
        if('error' in res.data){
            alert(res.data.error)
        }
        else {
            console.log(res.data);
            console.log('xx');

            localStorage.setItem("token", "true");
            window.location.reload(true); // refresh page
            history.push('/');
            console.log('reached this');
        }    
    })
    .catch((err:any) => {
        console.log(err.message, err.name)
    })
    // reset
    setLoginInput(initialInputs)
    }

return (
    <form onSubmit={handleSubmit}>
      <h2>Login</h2>
        <label>Username
          <input
            type='text'
            name='username'
            value={loginInput.username}
            onChange={handleLoginInput} 
          />
        </label>
        <label>Password
          <input
            type='password'
            name='password'
            value={loginInput.password}
            onChange={handleLoginInput} 
          />
        </label>
        <button>Log In</button>
      </form>
)
}

export default Login
