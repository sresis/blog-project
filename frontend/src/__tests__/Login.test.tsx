import React from "react";
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement } from "@testing-library/react";
import Login from "../Components/Login";
import Enzyme, { shallow, mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });
configure({adapter: new Adapter()});

function renderLogin() {
  
    return render(<Login/>);
  }

  
  describe('Check input fields', () => {
    it('There should be 2 input fields', () => {
        let test = (<Login/>);
        let wrapper = mount(test);
        expect(wrapper.find('input')).toHaveLength(2);
        expect(wrapper.find('input#username')).toHaveLength(1);

    });
  });