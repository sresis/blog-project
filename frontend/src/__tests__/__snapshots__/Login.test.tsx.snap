// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<form
  data-testid="login-form"
  id="login-form"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <label>
    Username
    <input
      data-testid="username"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Password
    <input
      data-testid="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </label>
  <button>
    Log In
  </button>
</form>
`;

exports[`Login form see if form renders 1`] = `
<form
  data-testid="login-form"
  id="login-form"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <label>
    Username
    <input
      data-testid="username"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Password
    <input
      data-testid="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </label>
  <button>
    Log In
  </button>
</form>
`;

exports[`Login info see if blank form renders 1`] = `
<form
  data-testid="login-form"
  id="login-form"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <label>
    Username
    <input
      data-testid="username"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Password
    <input
      data-testid="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </label>
  <button>
    Log In
  </button>
</form>
`;
